ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -213,17 +213,22 @@ protected String resolveUserToRunAs(Map<String, String> cache, Table t, Partitio
   @Override
   public void init(AtomicBoolean stop) throws Exception {
     super.init(stop);
+    initExecutorService();
```

Comment:
can we move the executor initialization into run method? 
````
public void run() {
  LOG.info("Starting Initiator thread");
  ExecutorService compactionExecutor = CompactorUtil.createExecutorWithThreadFactory(
      conf.getIntVar(HiveConf.ConfVars.HIVE_COMPACTOR_REQUEST_QUEUE),
      COMPACTOR_INTIATOR_THREAD_NAME_FORMAT);
  ....
}
````

Reply:
Done, fixed using `ExecutorService` with `try-with-resources.`
Starting from Java 19 `ExecutorService` implements `AutoCloseable` and this can be done.